settings.js

```diff
@@ -302,28 +302,35 @@ function findConflicts(schemas) {
 */
 var winprops = [];
 
-function winprop_match_p(meta_window, prop) {
-    let wm_class = meta_window.wm_class || "";
-    let title = meta_window.title;
-    if (prop.wm_class.constructor === RegExp) {
-        if (!wm_class.match(prop.wm_class))
-            return false;
-    } else if (prop.wm_class !== wm_class) {
-        return false;
-    }
-    if (prop.title) {
-        if (prop.title.constructor === RegExp) {
-            if (!title.match(prop.title))
+function _winprop_match_p_helper(user_attr_prop, _win_prop) {
+    // matches user-supplied conditions with null checking
+    if (user_attr_prop) {
+        let win_prop = String(_win_prop);
+        if (user_attr_prop.constructor === RegExp) {
```

Comment:
Checked, it does. Also this fixes the null issue without requiring an explicit null check, which actually keeps the semantics closer to the original code. I think I'm actually just going to apply this fix to the original code, because then it truly just is a fix, whereas actually everything written here so far changes the semantics.